lib/web_http.dart

```diff
@@ -91,13 +123,13 @@ class HttpClient {
     });
     request.onLoadEnd.listen((ProgressEvent event) {
       if (controller.isClosed) return;
-      if (request.status == 200) {
+      if (request.status >= 200 && request.status < 300) {
```

Comment:
I'm not an expert on HTTP codes, but aren't 3xx codes considered successful too?